### PR TITLE
Truncate file names while preserving file extensions

### PR DESCRIPTION
**Pull request by JammingBen**
## Related Issue

- https://github.com/owncloud/owncloud-design-system/issues/1758
- https://jira.owncloud.com/browse/WEB-1026

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

**Comment by JammingBen** [comment]
A few notes/questions:

* Do we want to separate the file name and the extension, e.g. by using the muted class on the extension? OC10 currently does this:
image
* Do we want to have tooltips on truncated file names? The issue I see here is that we can't determine if a file name is being truncated or not. So we'd have to implement tooltips for all file names.
* With very long folder paths, the filename will not be displayed at all. It was like this before, and I don't see an easy way to overcome this, but I want to point it out. It can be kinda confusing now with the file extension showing at the end.
image

@kulmann @pascalwengerter Any opinions here?

**Comment by fschade** [comment]
@JammingBen just an idea, if we have a path in the resource name, can we create 3 containers?
(a/very/long/pa...)(aLongFileNa....)(.ext)

**Comment by JammingBen** [comment]
Actually it is already divided in 3 containers. But as mentioned above, in the near future we will have the path beneath the file name (https://github.com/owncloud/web/issues/5953) anyway, so we problem becomes kinda obsolete IMO.

**Review by kulmann — CHANGES_REQUESTED** [review]

File: src/components/atoms/OcResourceName/OcResourceName.vue [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-      class="oc-resource-basename"
-      v-text="displayName"
-    /><span v-if="extension" class="oc-resource-extension" v-text="displayExtension" />
+    <span class="oc-text-truncate">
+      <span v-if="displayPath" class="oc-resource-path" v-text="displayPath" /><span
+        class="oc-resource-basename"
+        v-text="displayName"
+      />
+    </span>
+    <span v-if="extension" class="oc-resource-extension" v-text="displayExtension" />
```
[the remark this answers is not in the document]
Comment (kulmann) [reply src/components/atoms/OcResourceName/OcResourceName.vue:15]: FYI what we need at the same time, is, that actual whitespace in the filenames gets preserved. We had a solution in the past without `white-space: pre` and that trimmed all leading and trailing whitespace from the spans.

[the remark this answers is not in the document]
Comment (JammingBen) [reply src/components/atoms/OcResourceName/OcResourceName.vue:15]: Played around with it, but this does not seem to work :( Tried both via one line and CSS. I think we would need to implement some custom solution to make this work properly.

[the remark this answers is not in the document]
Comment (kulmann) [reply src/components/atoms/OcResourceName/OcResourceName.vue:15]: How'd you like to continue with this? the screenshot that I posted is too much of a regression in my opinion.

[the remark this answers is not in the document]
Comment (JammingBen) [reply src/components/atoms/OcResourceName/OcResourceName.vue:15]: I'll do another test with a custom after element. If that doesn't work, the only option I see is to implement a custom solution. I don't know yet how that could look like, maybe calculate the max number of characters based on the current width of the table cell?

[the remark this answers is not in the document]
Comment (tbsbdr) [reply src/components/atoms/OcResourceName/OcResourceName.vue:15]: i would go with 1) as well

**Review by kulmann — APPROVED** [review]

**Pull request merged**